example/src/main/tipc.ts

```diff
@@ -19,6 +19,11 @@ export const router = {
   readPkg: t.procedure.action(async () => {
     return fs.readFileSync("package.json", "utf-8")
   }),
+
+  readPkgStream: t.procedure.action(async function* () {
+    yield 1
+    yield "1"
+  }),
```

Comment:
If you delete the previous procedure methods, the type will work for this method.

I think you're hitting some recursion limit.